doc/decisions/binary.md

```diff
@@ -0,0 +1,40 @@
+# Binary
+
+## Problem
+
+binary
+
+- is the only metadata that allows more values (instead of making the validation stricter)
+- is the only API that modifies metadata
+
```

Comment:
Can you please explain the inconsistencies in more detail please? My proposal is to not let the API modify any metadata, why would it affect `KEY_FLAG_RO_META`?

Reply:
Your proposal does affect `KEY_FLAG_RO_META` in a positive way. The issue is with the current implementation, hence the note in the "Problems" section.

The name `KEY_FLAG_RO_VALUE` indicates that this flag determines whether a key's value is writable or not. However, this is not entirely true. For binary values `KEY_FLAG_RO_META` is also relevant, because we must modify the `meta:/binary` metadata.

The proposed solution doesn't have the problem, because metadata is not used anymore.